### PR TITLE
Issue86 91

### PR DESCRIPTION
**Pull request by haribalankumar**
Resolved issues 86 and 91. 
1. added field to be read in from ipdata file
2. if you have a ipdata file with tissue density and a surface exnode file, then split_data will split the ipdata into 5 exdata files - one for each lobe.

### CODE REVIEW

File: src/lib/geometry.f90 [hunk 1]
```diff
@@ -1341,6 +1497,407 @@ subroutine make_data_grid(surface_elems,spacing,to_export,filename,groupname)
...
+
+    sub_name = 'split_datacloud'
+    call enter_exit(sub_name,1)
+    
+    if(.not.allocated(globalnodelobe)) allocate(globalnodelobe(num_data))
+
+
+     if(lobe == "LLL") then 
+!      surfacelobe_elems = (/65,66,67,71,73,77,78,82,83,101,102,103,&
+!      104,106,107,108,109,111,112,113,114,115,117/)
```
Comment (alysclark) [inline src/lib/geometry.f90]: I get the need for hardcoded stuff here - for now :) - but we shouldn't have lines commented out - should be only one option, right?

File: src/lib/geometry.f90 [hunk 2]
```diff
@@ -2588,6 +3145,53 @@ function coord_at_xi(ne,xi,basis)
 
   end function coord_at_xi
 
+
+!
+!###################################################################################
+!
+  subroutine splitwords(line, words, nw )
```
Comment (alysclark) [inline src/lib/geometry.f90:5442]: Subroutine or function?

File: src/lib/geometry.f90 [hunk 3]
```diff
@@ -2588,6 +3145,53 @@ function coord_at_xi(ne,xi,basis)
...
+        nw = k
+        words( 1 : nw ) = buf( 1 : nw )
+    enddo
+
+  end subroutine splitwords
+
+
+!###################################################################################
+
+  subroutine words_to_real( words,ints, ni )
```
Comment (alysclark) [inline src/lib/geometry.f90:5464]: Subroutine or function? Belongs in Geometry or utilities?

File: src/lib/geometry.f90 (hunk 2 above)
Comment (alysclark) [reply src/lib/geometry.f90:5442]: Belongs in geometry or utilities?

File: src/lib/geometry.f90 [hunk 4]
```diff
@@ -41,13 +41,17 @@ module geometry
   public element_connectivity_1d
   public element_connectivity_2d
   public inlist
+  public split_datacloud
+  public split_datacloudfromSTL
   public evaluate_ordering
   public get_final_real
+  public splitwords
```
Comment (alysclark) [inline src/lib/geometry.f90:47]: WE only need to make things public if they are used elsewhere or will have bindings.

File: src/lib/geometry.f90 (hunk 3 above)
Comment (alysclark) [reply src/lib/geometry.f90:5464]: @tawhai please also comment on this philosophical point. In the geometry module we have a number of functions taht are doing things that essentially check numbers that are input. etc. in general these are functions not subroutines, but perhaps a greater question is do these belong in the geometry module at all - they have nothing to do really with geometry, and are utilities associated with this codebase, as far as I interpret them.

**Review by alysclark — CHANGES_REQUESTED** [review]

**Comment by tawhai** [comment]
I’m happy to go through the list below and make changes. BUT:

  *   I didn’t add any of the subroutines that are referred to below. I do not have them in my local code, i.e. split_datacloud, splitwords,  words_to_real).
  *   My commit for this subroutine mainly had formatting changes.

File: src/lib/geometry.f90 (hunk 3 above)
Comment (alysclark) [reply src/lib/geometry.f90:5464]: @tawhai i think i worded my comment a little vaguely here. I wasn't asking for you to make any changes, this is @haribalankumar's pull request, seperate from your recent one. Just wanted you to take a look at these new subroutines he has added, and comment on whether you think they are subroutines or functions, and whether this type of function (from a lungsim philosophy perspective) belongs in the geometry module? Sorry for not being clearer!

File: src/lib/geometry.f90 (hunk 1 above)
Comment (haribalankumar) [reply src/lib/geometry.f90]: yes . removed all unnecessary comments

File: src/lib/geometry.f90 (hunk 2 above)
Comment (haribalankumar) [reply src/lib/geometry.f90:5442]: Belongs to utilities. Will clean it up after raising another issue , lots of such functions/subroutines currently in geometry

**Comment by haribalankumar** [comment]
@alysclark I have made these changes requested. Raised a separate issue for cleaning up geometry.

**Comment by alysclark** [comment]
@haribalankumar  this request has conflicts with the main branch - can you merge this into your branch and resolve conflicts?

**Pull request merged**